JetAnalyzers/bin/jet_response_fitter_x.cc

```diff
@@ -255,6 +271,584 @@ int main(int argc,char**argv)
 ////////////////////////////////////////////////////////////////////////////////
 
 //______________________________________________________________________________
+TH1F *hXaxis( TH1F* hist, double minX, double maxX)
+{
+  double *xbins = new double [hist->GetNbinsX()+1];
+  int nbins{0};
+  for(int i=0; i<hist->GetNbinsX(); i++){
+    double x {hist->GetXaxis()->GetBinCenter(i+1)};
+    if(x>=minX && x<=maxX){
+      xbins[nbins]  = hist->GetXaxis()->GetBinLowEdge(i+1);
+      xbins[nbins+1]= hist->GetXaxis()->GetBinUpEdge(i+1);
+      ++nbins;
+    }
+  }
+
+  TH1F *hnew = new TH1F ("hnew", "", nbins, xbins);
```

Comment:
For xbins please use a vector<double>. You can even initialize a vector from an array.

Then when you pass it to the TH1F use &xbins.at(0). Also then nbins will be available from the array.

I just like C++ style more than C. I also have a personal vendetta against arrays. ;)